### ISSUE TITLE
Feature requests: "String manipulation"

### ISSUE THREAD

**Issue by sagivf** [rethinkdb/rethinkdb#5297]
I think some sort of `substring` api would be helpful.
`trim` would also be helpful.

Currently, I'm doing this:

`function rethinkSubString() {
  let url = r.row;
  let primaryKeyLength = 100;

  return r.branch(url.split('').count().gt(primaryKeyLength),
    url.match('^.{0}(.{' + primaryKeyLength + '})')('groups')(0)('str'),
    url
  );
}`

**Comment by danielmewes** [comment]
A substring function is going to be in the upcoming RethinkDB 2.3 through the `slice` command, see https://github.com/rethinkdb/rethinkdb/issues/4227

I'm going to mark this a ReQL proposal for the `trim` term that you're proposing. We'll discuss it for one of the upcoming releases.

**Comment by sagivf** [comment]
@danielmewes Any update on this? I don't see it on 2.3 or 2.4 unless I'm missing something?

**Comment by danielmewes** [comment]
@sagivf `slice` is already implemented, so it's going to ship with RethinkDB 2.3.0. https://github.com/rethinkdb/rethinkdb/issues/4227

I'm not sure when we'll ship the `trim` term. We first need to discuss and agree on the API, which is going to happen for one of the upcoming releases (we have a special "ReQL discussion period" at the beginning of each release cycle).

**Comment by sagivf** [comment]
@danielmewes Thanks, I got confused between 2.3.0 and 2.2.3, that's why I thought something was missing... Sorry about that.

Is there any target date for 2.3? What's the best place to follow - Github Millstones?

**Comment by danielmewes** [comment]
@sagivf Yeah, following the 2.3 Github milestone should give you an overview of what's left to do for 2.3.
The release is planned for March.
